### PR TITLE
DPL Analysis: event mixing: Move GroupSlicer to a separate file, add operator+ and +=

### PR DESCRIPTION
**Pull request by saganatt**
Moved GroupSlicer to avoid circular dependencies later in event mixing code.

### CODE REVIEW

File: Framework/Core/include/Framework/AnalysisTask.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "Framework/Expressions.h"
 #include "Framework/ExpressionHelpers.h"
 #include "Framework/EndOfStreamContext.h"
+#include "Framework/GroupSlicer.h"
```
Comment (ktf) [inline Framework/Core/include/Framework/AnalysisTask.h:24]: Do you really need to include it by default? Can't we include it only when needed?

File: Framework/Core/include/Framework/GroupSlicer.h [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+      GroupSlicerIterator copy = *this;
+      copy.position += inc;
+      copy.mGroupingElement = copy.mGroupingElement + inc;
+      return copy;
+    }
+
+    GroupSlicerIterator operator+=(int64_t inc)
+    {
+      position += inc;
+      mGroupingElement = mGroupingElement + inc;
```
Comment (aalkin) [inline Framework/Core/include/Framework/GroupSlicer.h]: Shouldn't it be `mGroupingElement += inc;`?

File: Framework/Core/include/Framework/AnalysisTask.h (hunk 1 above)
Comment (saganatt) [reply Framework/Core/include/Framework/AnalysisTask.h:24]: It is used in this file, for grouping for analysis task.

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
Needs forced sync merge with O2Physics after local test (ongoing)

**Pull request merged**